common/src/main/java/com/wynntils/features/user/overlays/map/MapFeature.java

```diff
@@ -32,12 +28,24 @@
 import com.wynntils.mc.render.Texture;
 import com.wynntils.mc.render.VerticalAlignment;
 import com.wynntils.mc.utils.McUtils;
+import com.wynntils.screens.maps.MainMapScreen;
 import com.wynntils.wynn.utils.WynnUtils;
 import net.minecraft.client.renderer.GameRenderer;
-import org.lwjgl.opengl.GL11;
-import org.lwjgl.opengl.GL13;
+import org.lwjgl.glfw.GLFW;
+
+public class MapFeature extends UserFeature {
+    public static MapFeature INSTANCE;
+
+    @RegisterKeyBind
+    public final KeyBind openMapKeybind = new KeyBind("Open Full Screen Map", GLFW.GLFW_KEY_M, false, () -> {
+        if (McUtils.mc().screen instanceof MainMapScreen mainMapScreen) {
```

Comment:
I'm trying to figure out how this works. Are we using the repeat delay from vanilla? So if we keep M pressed, after a while vanilla is going to send this as a repeated event, and if that is done when the map is displaying, we change mode to "close on key release"? Since we are not sending in `true` to the "just first press" boolean or whatever it was called?

If my guess is correct, maybe write a few lines about it? It is kind of magic otherwise.


I assume this was how Legacy also behaved, even though I never noticed or used it. It is very cool anyway, being able to quickly see the entire map just by holding M a bit longer.

Reply:
Added the comment.